bin/tx-prover/src/proxy/mod.rs

```diff
@@ -41,6 +50,17 @@ impl Worker {
     pub fn new(worker: Backend) -> Self {
         Self { worker, is_available: true }
     }
+
+    /// Update the worker configuration in the configuration file.
+    ///
+    /// # Errors
+    /// - If the worker address cannot be converted to [WorkerConfig].
+    fn update_worker_config(workers: &[Worker]) -> Result<(), String> {
+        let worker_configs =
+            workers.iter().map(|worker| worker.try_into()).collect::<Result<Vec<_>, _>>()?;
+
+        ProxyConfig::update_workers(worker_configs)
+    }
```

Comment:
Adding this method to the `Worker` struct feels a bit weird to me. Specifically, it is not very intuitive that interacting with a `Worker` would update `ProxyConfig`.

Given that this method is pretty simple and is used only once, I'd probably just get rid of it and move the code to its current callsite.